### ISSUE TITLE
scale command try to modify VNET address space

### ISSUE THREAD

**Issue by jalberto** [Azure/acs-engine#1714]
**Is this a request for help?**:
No
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of acs-engine?**:
0.9.1
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
1.7.7

**What happened**:
New scale up command try to modify existing VNET address space, it is not allowed in VNETs using peer to connect other VNETs

```

```

**What you expected to happen**:
To scaleup without modify VNET address space

**How to reproduce it** (as minimally and precisely as possible):
just run `scale` command

**Anything else we need to know**:

**Comment by jackfrancis** [comment]
@JackQuincy can you confirm that scaling will not work flawlessly in the described scenario?

**Comment by rocketraman** [comment]
I can confirm this issue -- tried the `scale` command with the latest master version and the deployment failed. CLI reports:

```
FATA[0016] resources.DeploymentsClient#CreateOrUpdate: Failure sending request: StatusCode=200 -- Original Error: Long running operation terminated with status 'Failed': Code="DeploymentFailed" Message="At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details." 
```

and the additional failure information in the Deployment in the portal contains:

```
{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.","details":[{"code":"BadRequest","message":"{\r\n \"error\": {\r\n \"code\": \"VnetAddressSpaceCannotChangeDueToPeerings\",\r\n \"message\": \"Address space of the virtual network /subscriptions/xxx/resourceGroups/my-Dev-Kube1/providers/Microsoft.Network/virtualNetworks/k8s-vnet-999 cannot change when virtual network has peerings.\",\r\n \"details\": []\r\n }\r\n}"}]}
```

**Comment by JackQuincy** [comment]
@jalberto To add a node to the VNET we need to be able to add an address for it in the VNET. I'm not sure how we could not. We do support bring your own VNET, where you specify your VNET when creating your cluster and we just join the cluster to that VNET.  Or am I misunderstanding your scenario?

**Comment by jackfrancis** [comment]
The expectation of "To scaleup without modify VNET address space" is not possible. Please re-open if there's additional documentation/information we can provide to unblock you!

**Comment by rocketraman** [comment]
> @jalberto To add a node to the VNET we need to be able to add an address for it in the VNET. I'm not sure how we could not. We do support bring your own VNET, where you specify your VNET when creating your cluster and we just join the cluster to that VNET. Or am I misunderstanding your scenario?

@JackQuincy @jackfrancis You guys are misunderstanding I believe... if you look at the error message I posted, the deployment is not trying to add an address to the VNET for the node, but rather change the address *space*. Specifically: VnetAddress*Space*CannotChangeDueToPeerings

**Comment by rocketraman** [comment]
Created a new issue for more visibility. I can't reopen this one.
